sdk/typespec/typespec_client_core/src/http/pipeline.rs

```diff
@@ -82,11 +84,22 @@ impl Pipeline {
         &self,
         ctx: &Context<'_>,
         request: &mut Request,
-    ) -> crate::Result<Response<T>> {
+    ) -> crate::Result<Response<T, DefaultFormat>> {
+        self.pipeline[0]
+            .send(ctx, request, &self.pipeline[1..])
+            .await
+            .map(|r| r.with_model_type())
+    }
+
+    pub async fn send_format<T, F>(
```

Comment:
I added this because _most_ cases will want to use the `DefaultFormat`. However, when a custom format **is** needed, the syntax gets clunky, because you'll have to call `.with_format()` on the result, but you can't do that until after you await. This results in having to write `Ok(self.pipeline.send(...).await?.with_format())` which felt clunky, because `with_format` is infallible but `send` is fallible. It works just fine, but if I could bundle it up into a single `send_format` call, then methods that create XML-formatted responses would follow a similar pattern as those that create JSON-formatted ones